### ISSUE TITLE
Align Embedded Logs data model with Standalone Logs data model

### ISSUE THREAD

**Issue by tigrannajaryan** [open-telemetry/opentelemetry-specification#622]
Now that we have an official standalone logs data model it will be beneficial to see if we can align logs embedded in the Spans (Span events) with this data model.

Embedded spans should use the same or similar model except the following which is already recorded in the containing Span:
- Trace context fields: trace id, span id, trace flags.
- Resource.

The differences currently are:
Span Event|Standalone Log Record|Notes
-----------|-----------------------|---------
Name|ShortName|Equivalent semantic, only field names are different.
-|Severity|Missing in Span Event.
-|Body|Missing in Span Event.
Attributes|Attributes|Almost equivalent, but Standalone Logs allow complex nested values that Span Events don't.

We need to decide which of these differences are acceptable/necessary. My preliminary thoughts are the following:
- Rename `Name` to `ShortName`. There is no semantics change.
- There is no need to add `Body` to Span Events. For standalone logs the `Body` is necessary to record what cannot be represented in the attributes or other fields for log formats that have unusual/unknown requirements. There are no such requirements for Span Events because we know they are generated (using OpenTelemetry API).
- We should extend Span Event `Attributes` to allow the same nested values. This [was requested/discussed](https://github.com/open-telemetry/opentelemetry-specification/issues/376) in the past too.
- I am torn on `Severity`. We should use it if we believe the [Error Semantics](https://github.com/open-telemetry/oteps/blob/master/text/0097-log-data-model.md) it introduces are applicable to Span Events.

**Comment by Oberon00** [comment]
* Body can be a semantic convention for attributes.
* Severity can be a semantic convention for attributes
* ShortName could be renamed to Name.
* Why are nested attributes needed for logs? Is this not cases where you could do JSON in / JSON out?

**Comment by tigrannajaryan** [comment]
>ShortName could be renamed to Name.

I would like to know what others think. This seems reasonable to me.

**Comment by tigrannajaryan** [comment]
>Body can be a semantic convention for attributes.

Yes, it can. The question is: do we need a Body for Span Events? Our API does not have anything like that and I do not think we need to add it to the API (but perhaps there is a good use case that I am not aware of).

**Comment by tigrannajaryan** [comment]
>Severity can be a semantic convention for attributes

I agree, it can. However, I think there is a value in consistency and uniformness. Making it a field in Log Data model and a semantic convention in Span Event is somewhat inconsistent.

**Comment by tigrannajaryan** [comment]
>Why are nested attributes needed for logs? 

This came up several times in the past in OpenTelemetry, not just for logs, but for Spans too. I think it is time to make it possible. 

>Is this not cases where you could do JSON in / JSON out?

We use strong typing for attribute values. There is no way to know that a string attribute is a JSON value that needs to be decoded to be understood.

**Comment by marcelocra-hubla** [comment]
Hi folks! Is this something you are still planning on doing? Tks.

**Comment by lmolkova** [comment]
Closing in favor of newer #4393 with additional context
